strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/event/artifact/MavenArtifactStoredEventListener.java

```diff
@@ -0,0 +1,49 @@
+package org.carlspring.strongbox.event.artifact;
+
+import org.carlspring.strongbox.repository.group.index.MavenIndexGroupRepositoryComponent;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.storage.repository.RepositoryLayoutEnum;
+
+import javax.inject.Inject;
+import java.io.IOException;
+
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class MavenArtifactStoredEventListener
+        extends BaseMavenArtifactEventListener
+{
+
+    @Inject
+    private MavenIndexGroupRepositoryComponent mavenIndexGroupRepositoryComponent;
+
+    @Override
+    public void handle(final ArtifactEvent event)
+    {
+        final Repository repository = getRepository(event);
+
+        if (!RepositoryLayoutEnum.MAVEN_2.getLayout().equals(repository.getLayout()))
```

Comment:
Please, avoid using the `RepositoryLayoutEnum.MAVEN_2.getLayout()`, as I'm thinking of removing this (you can check #437 for more details). Where possible, could you please use `Maven2LayoutProvider.ALIAS` instead...? Thanks!

I know, it's going to be tricky to remove this enum because it's used in places like `Repository` for the default layout.